app/code/Magento/Persistent/Helper/Data.php

```diff
@@ -193,6 +202,8 @@ public function getPersistentConfigFilePath()
      */
     public function canProcess($observer)
     {
-        return true;
+        return null === $this->storeManager ? 
+            true :
+            $this->isEnabled($this->storeManager->getStore()->getCode());
```

Comment:
I think there are no cases when store manager might be null
```suggestion
        return $this->isEnabled($this->storeManager->getStore()->getCode());
```